src/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -178,24 +178,34 @@ public static unsafe void Clear(Array array, int index, int length)
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
 
-            ref byte p = ref GetRawArrayGeometry(array, out uint numComponents, out uint elementSize, out int lowerBound, out bool containsGCPointers);
+            ref byte p = ref Unsafe.As<RawArrayData>(array).Data;
+            int lowerBound = 0;
+
+            MethodTable* pMT = RuntimeHelpers.GetMethodTable(array);
+            if (pMT->IsMultiDimensionalArray)
+            {
+                int rank = pMT->MultiDimensionalArrayRank;
+                lowerBound = Unsafe.Add(ref Unsafe.As<byte, int>(ref p), rank);
+                p = ref Unsafe.Add(ref p, 2 * sizeof(int) * rank); // skip the bounds
```

Comment:
Does the _rank_ local need to be a `size_t` to prevent integer overflow in this multiplication?

Reply:
It does not need to be `size_t`. The maximum rank is 32.